### PR TITLE
feat(ui): Supporting subtypes for data jobs

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

3 important changes:

1. Support subtypes for data jobs
2. Fix preview rendering for charts with subtypes. 
3. Clean up chart + dataset subtype mapping to avoid using additional db lookups. 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx [hunk 1]
```diff
@@ -68,6 +70,8 @@ export const Preview = ({
             urn={urn}
             description={description || ''}
             type="Data Task"
+            entityType={EntityType.DataJob}
```
Comment (asikowitz) [inline datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx:73]: Might have to merge the refactor I did around here first...

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx [hunk 2]
```diff
@@ -168,6 +168,7 @@ export class ChartEntity implements Entity<Chart> {
         return (
             <ChartPreview
                 urn={data.urn}
+                subType={data.subTypes?.typeNames?.[0]}
```
Comment (asikowitz) [inline datahub-web-react/src/app/entity/chart/ChartEntity.tsx:171]: Did we already do charts?

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java [hunk 3]
```diff
@@ -1174,10 +1174,6 @@ private void configureDatasetResolvers(final RuntimeWiring.Builder builder) {
                 .dataFetcher("testResults", new TestResultsResolver(entityClient))
                 .dataFetcher("aspects", new WeaklyTypedAspectsResolver(entityClient, entityRegistry))
                 .dataFetcher("exists", new EntityExistsResolver(entityService))
-                .dataFetcher("subTypes", new SubTypesResolver(
```
Comment (asikowitz) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java:1177]: Are we trying to move off of this file and into the mappers?

**Review by asikowitz — COMMENTED** [review]
Main question is around the refactor I did for `Preview.tsx`, passing in entity type as `type`

File: datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx:73]: please send me the commit

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx (hunk 2 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/chart/ChartEntity.tsx:171]: yeah charts are done, this is just a fix for a missing prop

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java (hunk 3 above)
Comment (jjoyce0510) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java:1177]: yes this incurs additional db calls

File: datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx:73]: thats part of a bigger refactoring you did.

for now, i'm going to keep the current convention until we decide how to proceed here

Comment (asikowitz) [reply datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx]: I think we should merge that in; I can work on that myself later

File: datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx [hunk 4]
```diff
@@ -68,6 +70,7 @@ export const Preview = ({
             urn={urn}
             description={description || ''}
             type="Data Task"
+            type={subType}
```
Comment (asikowitz) [inline datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx:73]: This looks like we're overwriting the `type` prop, think this should be something like `subtype || 'Data Task'`

**Review by asikowitz — APPROVED** [review]
One comment I'd like addressed before merge, otherwise LGTM

File: datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx (hunk 4 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/dataJob/preview/Preview.tsx:73]: thanks for catching this!

**Pull request merged**